src/Microsoft.Identity.Client/ApiConfig/GetAuthorizationRequestUrlParameterBuilder.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Identity.Client.ApiConfig
+{
+    /// <summary>
+    ///     NOTE:  a few of the methods in AbstractAcquireTokenParameterBuilder (e.g. account) don't make sense here.
+    ///     Do we want to create a further base that contains ALL of the common methods, and then have another one including
+    ///     account, etc
+    ///     that are only used for AcquireToken?
+    /// </summary>
+    public sealed class GetAuthorizationRequestUrlParameterBuilder :
+        AbstractAcquireTokenParameterBuilder<GetAuthorizationRequestUrlParameterBuilder, IGetAuthorizationRequestUrlParameters>
+    {
+        /// <summary>
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns></returns>
+        internal static GetAuthorizationRequestUrlParameterBuilder Create(IEnumerable<string> scopes)
+        {
+            return new GetAuthorizationRequestUrlParameterBuilder().WithScopes(scopes);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="redirectUri"></param>
+        /// <returns></returns>
+        public GetAuthorizationRequestUrlParameterBuilder WithRedirectUri(string redirectUri)
```

Comment:
I don't understand this scenario? we would not use the RedirectUri set in the CCA?

Reply:
ConfidentialClientApplication.cs today has this method:

        public async Task<Uri> GetAuthorizationRequestUrlAsync(IEnumerable<string> scopes, string redirectUri, string loginHint,
            string extraQueryParameters, IEnumerable<string> extraScopesToConsent, string authority)

It takes a redirectUri so I have that ability here.  Should we not have it for this case?

---
In reply to: [246268634](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/766#discussion_r246268634) [](ancestors = 246268634)